native/core/src/execution/jni_api.rs

```diff
@@ -319,6 +320,7 @@ fn parse_memory_pool_config(
             "greedy_global" => MemoryPoolConfig::new(MemoryPoolType::GreedyGlobal, pool_size),
             "fair_spill" => MemoryPoolConfig::new(MemoryPoolType::FairSpill, pool_size_per_task),
             "greedy" => MemoryPoolConfig::new(MemoryPoolType::Greedy, pool_size_per_task),
+            "unbounded" => MemoryPoolConfig::new(MemoryPoolType::Unbounded, 0),
```

Comment:
TBH, I don't know if it is ever a good idea to allow an unbounded memory pool. It doesn't hurt to have the option, but under what conditions is this choice useful?

Reply:
I guess we can add this advice in the documentation. Speeding up the job sounds like a win until the jobs start OOMing. :)

Reply:
I see the value in having the unbounded tool for development testing purposes but not for end user use.

I would like to use it to help with https://github.com/apache/datafusion-comet/issues/1315